src/main/java/seedu/trackermon/logic/parser/SortCommandParser.java

```diff
@@ -31,48 +47,143 @@ public SortCommand parse(String args) throws ParseException {
+        return new SortCommand(comparator);
+    }
+
+    /**
+     * Returns a comparator then chain the previous comparator to a new one if the previous is not null.
+     * @param comparator previous comparator to build upon.
+     * @param entry contain the new comparator.
+     * @return a comparator build upon the previous comparator.
+     */
+    private static Comparator<Show> chainComparator(Comparator<Show> comparator,
+                                                    Map.Entry<Comparator<Show>, Integer> entry) {
+        if (comparator == null) {
+            return entry.getKey();
+        } else {
+            return comparator.thenComparing(entry.getKey());
         }
+    }
 
-        //only name present
-        if (isNamePresent && !isStatusPresent) {
-            return new SortCommand(nameComparator);
+    /**
+     * Returns a new comparator based on the values in sorted Map.
+     * @param sortedOrder the sorted map for the new comparator to build upon.
+     * @return a new comparator based on the values in sorted Map.
+     */
+    private static Comparator<Show> buildComparator(Map<Comparator<Show>, Integer> sortedOrder) {
+        Comparator<Show> overallComparator = null;
+        for (Map.Entry<Comparator<Show>, Integer> entry: sortedOrder.entrySet()) {
+            if (entry.getValue() != NO_VALUE) {
+                overallComparator = chainComparator(overallComparator, entry);
+            }
         }
+        return overallComparator;
+    }
 
-        //only status present
-        if (isStatusPresent && !isNamePresent) {
-            return new SortCommand(statusComparator);
+    /**
+     * Checks if the prefix is used and if it is, update Order map.
+     * Checks if descending value is used correctly, update the Order map with descending comparator and
+     * returns the comparator.
+     * @param comparator key of the Order Map.
+     * @param prefix the Prefix to search.
+     * {@code ArgumentMultimap}.
+     * @return a comparator depending on whether it was reversed.
+     */
+    private static Comparator<Show> putIntoMap(Comparator<Show> comparator,
+                                               Prefix prefix, ArgumentMultimap argMultimap) {
+        if (arePrefixesPresent(argMultimap, prefix)) {
+            String valueName = argMultimap.getValue(prefix).orElse(VALUE_ASC);
+            valueName = valueName.toUpperCase().trim();
+            if (valueName.equals(VALUE_DSC)) {
+                comparator = comparator.reversed();
+            }
+            order.put(comparator, startingValue);
+            startingValue++;
         }
 
-        //both name and status present - status first then name
-        if (arePrefixesPresent(argMultimap, PREFIX_SORT_STATUS_ORD)) {
-            return new SortCommand(statusComparator.thenComparing(nameComparator));
+        return comparator;
+    }
+
+    /**
+     * Updates value of Order Map, if the value is not -1.
```

Comment:
I think the comma in this line is redundant.

Reply:
ok